### PR TITLE
Show Cashier What's Up on Customer Display

### PR DESCRIPTION
**Pull request by blubaugha**
### Issues Fixed
[PCOM-3612](https://jira.ae.com/browse/PCOM-3612)
[PCOM-3631](https://jira.ae.com/browse/PCOM-3631)

### Summary
This updates the **Loyalty** button, on the **Sale** and **Return** screens, to show the cashier insulting messages about the customer's signup status on the **Customer Display**.

### Screenshots
**Left** - Associate Display
**Right** - Customer Display

**Loyalty Signup**
loyalty-signup

**Verify Customer Information**
verify-customer-info

**Make Corrections to Customer Information**
make-corrections

**Terms and Conditions**
terms-and-conditions

**Returns Screen Behaves Like Sale Screen**
returns

**Cancel**
cancel

**Opt-Into Marketing and Finish**
marketing-opt-in

### CODE REVIEW

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/services/animation.service.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    public restart(selector: string | Element, animationClass: string): void {
+        if (!selector) {
+            return;
+        }
+
+        let element = typeof selector === 'string' ? document.querySelector(selector) : selector;
+
+        if (element) {
+            this.renderer.removeClass(element, animationClass);
+            // A trick to trigger a browser browser reflow layout calculations, which will cause animations to run
```
Comment (jchirico8) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/services/animation.service.ts:23]: Kids on Halloween: Trick or Treat!
Andy: Here's a neat trick to get the browser to rerun animations!!

**Comment by stojankz** [comment]
This might be unrelated to the code you've changed, but as a general comment/opinion that Cancel (button?) is taking up quite a bit of vertical space.. Any way that could be smaller text or have the cancel be handled by an X in the top right of that mini info panel?

**Review by jchirico8 — COMMENTED** [review]
Looks good besides the GIANT cancel font

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/directives/animation.directive.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { Directive, ElementRef, Input, OnChanges, OnDestroy, OnInit, Renderer2, SimpleChanges } from '@angular/core';
+import { Observable } from 'rxjs/internal/Observable';
+import { AnimationService } from '../../core/services/animation.service';
+import { isObservable, merge, Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+
+@Directive({
```
Comment (dankaste) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/directives/animation.directive.ts:8]: Whats the reason for not using the animations property in the component decorator? Doesn't that take care of most of this for you?

Comment (blubaugha) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/directives/animation.directive.ts:8]: This was easier for me 😄 
But I'll give Angular animations a try and see if it works out mOrE bEtTeR

**Comment by blubaugha** [comment]
@stojankz 
> This might be unrelated to the code you've changed, but as a general comment/opinion, that Cancel (button?) is taking up quite a bit of vertical space.. Any way that could be smaller text or have the cancel be handled by an X in the top right of that mini info panel?

@jchirico8 
> Looks good besides the GIANT cancel font

# Updated to not have a

##### massive

# cancel button

### Lets me knows what you thinks

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/directives/animation.directive.ts (hunk 2 above)
Comment (blubaugha) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/directives/animation.directive.ts]: I rewrote the CSS animations the Angular way, but I think it's a lot more complex like this. But, we can leave it this if you want it more in alignment with how Angular does it.

**Review by chenson42 — APPROVED** [review]

**Pull request merged**